pkg/scheduler/framework/plugins/helper/spread.go

```diff
@@ -36,36 +50,43 @@ func DefaultSelector(pod *v1.Pod, sl corelisters.ServiceLister, cl corelisters.R
 			labelSet = labels.Merge(labelSet, service.Spec.Selector)
 		}
 	}
+	selector := labelSet.AsSelector()
 
-	if rcs, err := cl.GetPodControllers(pod); err == nil {
-		for _, rc := range rcs {
-			labelSet = labels.Merge(labelSet, rc.Spec.Selector)
-		}
+	owner := metav1.GetControllerOfNoCopy(pod)
+	if owner == nil {
+		return selector
 	}
 
-	selector := labels.NewSelector()
-	if len(labelSet) != 0 {
-		selector = labelSet.AsSelector()
+	gv, err := schema.ParseGroupVersion(owner.APIVersion)
+	if err != nil {
+		return selector
```

Comment:
one corner case here, a bare pod created after the a matching and existing RS, it was not owned by any controller. pods owned by the RS will get counted in the old logic but not true now.

Reply:
In that case the replicaset controller is supposed to adopt the pod and update its owner reference. A race could happen (the pod gets scheduled before the replicaset controller gets to update the owner reference), but I don't believe this is a common case that we should care about, especially it has a non-negligible impact on performance.

Reply:
Two cases here, 
- bare pod created before the RS, the pod will be acquired by the RS
- bare pod created after the RS, RS won't acquire the pod.

So, should the code fall back to the old logic when the ownerreference is not set?

Reply:
In all cases the replicaset controller will try to find a replicaset to adopt the pod, it may well be the case that the scheduler will try to schedule the pod before it gets adopted. 

But what is the use case for the two cases you mentioned? If there is no legitimate case, then I don't think we should complicate our code to support it.

Reply:
no problem, just thought the diff might not nil.